examples/publish_course_manual_rb/publish_manual.rb

```diff
@@ -0,0 +1,176 @@
+require "rest-client"
+require "mime/types"
+require "hashie/mash"
+require "base64"
+require "json"
+require "httplog"
+require "slop"
+require "active_support/core_ext/object/blank"
+
+class ManualPublisher
+  POLL_SLEEP_SECS = 5
+  MAX_POLLS = 20
+
+  attr_reader :course_id, :attachment_dir, :html_file, :api_hostname, :api_token, :api_secret
+
+  def initialize(course_id:, attachment_dir:, html_file:, api_hostname:, api_token:, api_secret:, verbose: false)
+    @course_id = course_id
+    @attachment_dir = attachment_dir
+    @html_file = html_file
+    @api_hostname = api_hostname
+    @api_token = api_token
+    @api_secret = api_secret
+
+    HttpLog.configure {|c| c.enabled = !!verbose }
+  end
+
+  def publish
+    puts "Deleting old draft..."
+    request(method: :delete)
+
+    puts "Uploading content..."
+    request(
+      method: :put,
+      payload: {
+        status: "content_complete",
+        content:,
+        attachment_upload_data: initial_attachment_upload_data&.to_json
+      }
+    )
+
+    if initial_attachment_upload_data
+      poll_for_state "needs_attachments", "draft"
+
+      puts "Uploading attachments..."
+      upload_attachments
+    end
+
+    request(method: :put, payload: { status: "ready_to_process" })
+    poll_for_state "ready_to_publish", "needs_attachments"
+
+    puts "Marking as published..."
+    request(method: :put, payload: { status: "published" })
+
+    puts "Done."
+  end
+
+  private
+
+  def upload_attachments
+    request.attachment_upload_data.each do |uploadable|
+      next unless uploadable.blob_upload_url
+
+      file_info = initial_attachment_upload_data_for(uploadable.file_path)
+      RestClient.put(
+        uploadable.blob_upload_url,
+        File.open(uploadable.file_path),
+        {
+          "Accept" => "application/json",
+          "Content-Type" => file_info[:content_type],
+          "Content-MD5" => file_info[:checksum],
+          "Content-Length" => file_info[:byte_size]  
+        }
+      )
+    end
+  end
+
+  def poll_for_state(desired_state, failure_state)
+    1.upto(MAX_POLLS) do
+      current_state = request&.state
+
+      if desired_state == current_state
+        puts "Now #{current_state}"
+        return
+      elsif failure_state == current_state
+        raise "Failure state of #{current_state} occurred"
+      else
+        puts "Currently #{current_state}, waiting for #{desired_state}"
+        sleep POLL_SLEEP_SECS
+      end
+    end
+
+    raise "Timed out waiting for #{desired_state}"
+  end
+
+  def request(method: :get, payload: nil)
+    r = RestClient::Request.execute(
+      method:,
+      url:,
```

Comment:
Cool seeing you use this syntax, but we also need to remember that this is a public script, so we need to note that this script requires Ruby 3.1+ if we use it.
Best way to do that is probably in the Gemfile with:
`ruby '>= 3.1.0'`